.github/workflows/main.yml

```
@@ -0,0 +1,40 @@
+name: QA
+on: [pull_request]
```

You probably meant to only run this workflow when there is code change, currently it would run for all of those event types: https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request

```suggestion
on:
  pull_request:
    types:
      - synchronize
      - opened
```